### PR TITLE
feat: new status effect manager

### PR DESCRIPTION
**Pull request by corp-0**
### Purpose
The Status Effect Manager is a little component that exposes an interface to add and handle effects on players (and maybe other stuff later?). 

StatusEffects are scriptable objects that can be composed using the StatusEffect interfaces such as ``IExpirableStatusEffect``. The manager knows this composition and acts accordingly.

As a proof of concept, the convulsing effect has been added and is triggered by electrocution.

### Notes:
Known issues:
- The convulsing effect triggers action text and the twitching emote. Problem is twitching emote also adds action text and they both overlap instead of queueing.
- The process to use EmoteActionManager is convoluted.
- Haven't tested triggering the emote over the network.

### Changelog:
CL:[New] Added convulsing status effect when electrocuted.

### CODE REVIEW

**Comment by corp-0** [comment]
@MaxIsJoe take a look here plox

**Comment by corp-0** [comment]
I will add docs once the thing is reviewed and polished

**Review by Bod9001 — DISMISSED** [review]
I assume its server side?

**Comment by corp-0** [comment]
> I assume its server side?

It is supposed to be interacted with from server side, but I'm not doing any network validations. I'm very rusty when it comes to our network system.

**Comment by corp-0** [comment]
The idea here is make some foundations and standardise our statuses and hopefully make everything run through this manager (drunk, confused, stunned, high, etc)

**Comment by MaxIsJoe** [comment]
> @MaxIsJoe take a look here plox

amazing, will defiantly use it to add some flavor to some emotes or interactions.

**Comment by PerfectTangent** [comment]
> The idea here is make some foundations and standardise our statuses and hopefully make everything run through this manager (drunk, confused, stunned, high, etc)

yeah, we should take a look at what effects we can generalize and move into this manager. Really, great work.

**Comment by Bod9001** [comment]
good too merge?

**Comment by corp-0** [comment]
No, I have some numbers to tweak to be closer to SS13 numbers. Anything you want to comment on the code itself? I was expecting criticism

File: UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		public override void DoEffect()
+		{
+			if (target.TryGetComponent<PlayerScript>(out var player) == false)
+			{
+				return;
+			}
+
+			if (DMMath.Prob(actualProcChance))
+			{
+				DropItemFromHands(NamedSlot.leftHand, player);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs:90]: Where does this come from?

Comment (PerfectTangent) [reply UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs:90]: The DMMath operation?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs:90]: it is the method to test chances we are supposed to use. You give it a percent and it returns a bool

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs:90]: NamedSlot

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/Convulsing.cs:90]: oh, it's enum don't worry I'm just dumb

**Comment by corp-0** [comment]
again ready for review. I added a couple unit tests to make sure the thing does what it is meant to do (and it helped me correcting stuff).

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**